pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go

```diff
@@ -784,6 +784,56 @@ func IsDuringAPIServerProgressingOnSNO(topology string, events monitorapi.Interv
 	}
 }
 
+func IsExcludedContainerBackoffRestart(topology string, event monitorapi.Interval) bool {
+	// This is a list of known container back-off restart failures
+	// Our goal is to conquer these restarts.
+	// So we are sadly putting these as exceptions.
+	// If you discover a container backoff restarting more than 3 times, it is a bug and you should investigate it.
+	type exceptionVariants struct {
+		containerName      string
+		topologyToExclude  string
+		namespaceToExclude string
+	}
+	exceptions := []exceptionVariants{
+		{
+			// https://issues.redhat.com/browse/OCPBUGS-45071
+			containerName:      "container/manager",
+			topologyToExclude:  "single",
+			namespaceToExclude: "openshift-operator-controller",
+		},
+		{
+			// https://issues.redhat.com/browse/OCPBUGS-45071
+			containerName:      "container/manager",
+			topologyToExclude:  "single",
+			namespaceToExclude: "openshift-catalogd",
+		},
+	}
+
+	for _, val := range exceptions {
+		matched, err := regexp.MatchString(val.containerName, event.Locator.OldLocator())
```

Comment:
Curious on the expectation here.  Testing against [4.18-e2e-aws-upgrade-ovn-single-node/1861099582801317888](https://prow.ci.openshift.org/view/gs/test-platform-results/logs/periodic-ci-openshift-release-master-ci-4.18-e2e-aws-upgrade-ovn-single-node/1861099582801317888) interval [events](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/logs/periodic-ci-openshift-release-master-ci-4.18-e2e-aws-upgrade-ovn-single-node/1861099582801317888/artifacts/e2e-aws-upgrade-ovn-single-node/single-node-e2e-test/artifacts/junit/events_used_for_junits_20241125-182913.json)

I get an OldLocator value of `namespace/openshift-catalogd node/ip-10-0-5-55.us-west-2.compute.internal pod/catalogd-controller-manager-59d9788859-75t2z hmsg/b20dd58ef6`

For
```
{
  "level": "Warning",
  "source": "KubeEvent",
  "display": true,
  "locator": {
    "type": "Kind",
    "keys": {
      "hmsg": "b20dd58ef6",
      "namespace": "openshift-catalogd",
      "node": "ip-10-0-5-55.us-west-2.compute.internal",
      "pod": "catalogd-controller-manager-59d9788859-75t2z"
    }
  },
  "message": {
    "reason": "BackOff",
    "cause": "",
    "humanMessage": "Back-off restarting failed container manager in pod catalogd-controller-manager-59d9788859-75t2z_openshift-catalogd(2aa86518-bb5e-4dc9-9a7a-8f1fa97a1aad)",
    "annotations": {
      "count": "66",
      "firstTimestamp": "2024-11-25T18:12:58Z",
      "interesting": "true",
      "lastTimestamp": "2024-11-25T18:40:50Z",
      "pathological": "true",
      "reason": "BackOff"
    }
  },
  "from": "2024-11-25T18:40:50Z",
  "to": "2024-11-25T18:40:51Z"
}


```

Which doesn't match `containerName: "container/manager"`

Reply:
That's a good point, it doesn't seem to call out the container in the locator. Maybe it makes more sense to key off the pod name? What do you think?

Reply:
@neisw I would much rather use something pre-existing so I will switch to this approach